docs/concepts/configuration/pod-priority-preemption.md

```diff
@@ -0,0 +1,197 @@
+
+**Note 3:** If you delete a PriorityClass, existing pods that use the name of the
+deleted priority class will remain unchanged, but you will not be able to create more pods
+that use the name of the deleted priority class.
+
+#### Example PriorityClass
+```yaml
+apiVersion: v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+
+## Pod Priority
+Once you have one or more PriorityClasses, you can create pods which specify one
+of those PriorityClass names in their spec. Priority admission controller uses
+`priorityClassName` field and populates the integer value of priority. If the priority
+class is not found, the pod will be rejected.
+
+The following YAML is an example of a pod configuration that uses the PriorityClass
+created above. Priority admission controller checks the spec and resolves the
+priority of the pod to 1000000.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    imagePullPolicy: IfNotPresent
+  priorityClassName: high-priority
+```
+
+## Preemption
+When pods are created, they go to a queue and wait to be scheduled. Scheduler picks a pod
+from the queue and tries to schedule it on a node. If no node is found that satisfies
+all the specified requirements (predicates) of the pod, preemption logic is triggered 
+for the pending pod. Let's call the pending pod P.
+Preemption logic tries to find a node where removal of one or more pods with lower priority 
+than P would enable P to schedule on that node. If such a node is found, one or more lower priority pods will
+be deleted from the node. Once the pods are gone, P may be scheduled on the node. 
+
+### Limitations of Preemption (alpha version)
+
+#### Starvation of Preempting Pod
+When pods are preempted, the victims get their
+[graceful termination period](https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods).
+They have that much time to finish their work and exit. If they don't, they will be
+killed. This graceful termination period creates a time gap between the point that
+scheduler preempts pods until the pending pod (P) can be scheduled on the node (N).
+In the meantime, scheduler keeps scheduling other pending pods. When one or more victims
+exit or get terminated, scheduler may place other pending pods on the node if the pods are
+ahead of P in the scheduling queue. In such a case, it is likely that
+when all victims exit, pod P won't fit on node N anymore. So, scheduler will have to
+preempt other pods on node N or another node to let P schedule. This scenario may 
+be repeated again for the second and subsequent rounds of preemption and P may not
+get scheduled for a while. This scenario can cause problems in various clusters, but
+is particularly problematic in clusters where many new pods are created all the time.
```

Comment:
Maybe say "in clusters with a high preemption rate" rather than "in clusters where many new pods are created all the time" as I think it's more related to the former?

Reply:
High creation rate is more of a problem that preemption rate. I changed it to "in clusters with a high creation rate". When creation rate is high, even a single preemption may face this issue.